components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/dao/impl/FileBasedUserStoreDAOImpl.java

```diff
@@ -426,4 +436,69 @@ private UserStoreDTO getUserStoreDTO(RealmConfiguration secondaryRealmConfigurat
         }
         return userStoreDTO;
     }
+
+    /**
+     * To check whether the provided domain name is valid or not.
+     *
+     * @param domainName user store domain name
+     * @param isAdd      boolean value, this will be true when add a new user store
+     * @return true or false
+     * @throws IdentityUserStoreMgtException
+     */
+    private boolean isValidDomain(String domainName, Boolean isAdd) throws IdentityUserStoreMgtException {
```

Comment:
better to have separate methods to check the domain name
- is valid to add
- is existing

Reply:
Fix with e9a9db12c05138fc6a0b5a93bfa691091189a8e1